src/renderer.zig

```diff
@@ -611,6 +616,171 @@ pub const MenuBackGround = struct {
 	}
 };
 
+pub const Skybox = struct {
+	var starShader: Shader = undefined;
+	var starUniforms: struct {
+		mvp: c_int,
+		starOpacity: c_int,
+	} = undefined;
+
+	var starVao: c_uint = undefined;
+
+	var starSsbo: graphics.SSBO = undefined;
+
+	const numStars = 10000;
+
+	fn getStarPos(seed: *u64) Vec3f {
+		const x: f32 = @floatCast(main.random.nextFloatGauss(seed));
+		const y: f32 = @floatCast(main.random.nextFloatGauss(seed));
+		const z: f32 = @floatCast(main.random.nextFloatGauss(seed));
+
+		const r = std.math.cbrt(main.random.nextFloat(seed))*5000.0;
+
+		return vec.normalize(Vec3f{x, y, z})*@as(Vec3f, @splat(r));
+	}
+
+	fn getStarColor(temperature: f32, light: f32, image: graphics.Image) Vec3f {
+		const rgbCol = image.getRGB(@intFromFloat(std.math.clamp(temperature/15000.0*@as(f32, @floatFromInt(image.width)), 0.0, @as(f32, @floatFromInt(image.width - 1)))), 0);
+		var rgb: Vec3f = @floatFromInt(Vec3i{rgbCol.r, rgbCol.g, rgbCol.b});
+		rgb /= @splat(255.0);
+
+		rgb *= @as(Vec3f, @splat(light));
+
+		const m = @reduce(.Max, rgb);
+		if(m > 1.0) {
+			rgb /= @as(Vec3f, @splat(m));
+		}
+
+		return rgb;
+	}
+
+	fn init() void {
+		const starColorImage = graphics.Image.readFromFile(main.stackAllocator, "assets/cubyz/star.png") catch |err| {
+			std.log.err("Failed to load star image: {s}", .{@errorName(err)});
+			return;
+		};
+		defer starColorImage.deinit(main.stackAllocator);
+
+		starShader = Shader.initAndGetUniforms("assets/cubyz/shaders/skybox/star.vs", "assets/cubyz/shaders/skybox/star.fs", "", &starUniforms);
+		starShader.bind();
+
+		var starData: [numStars*20]f32 = undefined;
+
+		const starDist = 200.0;
+
+		const off: f32 = @sqrt(3.0)/6.0;
+
+		const triVertA = Vec3f{0.5, starDist, -off};
+		const triVertB = Vec3f{-0.5, starDist, -off};
+		const triVertC = Vec3f{0.0, starDist, @sqrt(3.0)/2.0 - off};
+
+		var seed: u64 = 0;
+
+		for(0..numStars) |i| {
+			var pos: Vec3f = undefined;
+
+			var radius: f32 = undefined;
+
+			var temperature: f32 = undefined;
+
+			var light: f32 = 0;
+
+			while(light < 0.1) {
+				pos = getStarPos(&seed);
+
+				radius = @floatCast(main.random.nextFloatExp(&seed)*4 + 0.2);
+
+				// 5772 is the temperature of the sun, so I divide by it to get the temperature in solar temperature
```

Comment:
I don't see a reason not to just keep it in Kelvin.
Since the `4000` below now is just an empirical constant, you can just readjust it bases on (1/5572)⁴
This would allow also removing the `*5772.0` below